src/main/java/com/uid2/admin/managers/S3KeyManager.java

```diff
@@ -0,0 +1,108 @@
+package com.uid2.admin.managers;
+
+import com.uid2.admin.store.writer.S3KeyStoreWriter;
+import com.uid2.shared.auth.OperatorKey;
+import com.uid2.shared.model.S3Key;
+import com.uid2.shared.secret.IKeyGenerator;
+import com.uid2.shared.store.reader.RotatingS3KeyProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.util.*;
+
+public class S3KeyManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(S3KeyManager.class);
+
+    private final RotatingS3KeyProvider s3KeyProvider;
+    private final S3KeyStoreWriter s3KeyStoreWriter;
+    private final IKeyGenerator keyGenerator;
+
+
+    public S3KeyManager(RotatingS3KeyProvider s3KeyProvider, S3KeyStoreWriter s3KeyStoreWriter, IKeyGenerator keyGenerator) {
+        this.s3KeyProvider = s3KeyProvider;
+        this.s3KeyStoreWriter = s3KeyStoreWriter;
+        this.keyGenerator = keyGenerator;
+
+
+    }
+
+    public S3Key generateS3Key(int siteId, long activates, long created) throws Exception {
+        int newKeyId = getNextKeyId();
+        S3Key newS3Key = new S3Key();
+        newS3Key.setId(newKeyId);
```

Comment:
Can we pass these fields through the constructor and remove the setters? Unless there will be a reason to change the values later. Key_id and site_id at least should not be updatable after creation.

Reply:
right, I will keep the setter for activation and remove others